src/app/shared/shared.module.ts

```diff
@@ -0,0 +1,31 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { HeaderComponent } from './components/header/header.component';
+import { ButtonComponent } from './components/button/button.component';
+import { InfoComponent } from './components/info/info.component';
+import { SearchComponent } from './components/search/search.component';
+import {FontAwesomeModule} from "@fortawesome/angular-fontawesome";
+import {CourseComponent} from "../course/course.component";
+
+
+
+@NgModule({
+  declarations: [
+    HeaderComponent,
+    ButtonComponent,
+    InfoComponent,
+    SearchComponent,
+    CourseComponent
```

Comment:
Since in shared module this list and exports list are same, we can create variable for it as it's mentioned in the second point of the task.